spec/mock/random-mock.js

```diff
@@ -0,0 +1,34 @@
+"use strict"
+
+var random;
+
+random = jasmine.createSpyObj("random", [
+    "bufferAsync",
+    "password"
+]);
+
+random.bufferAsync = jasmine.createSpy("random.bufferAsync").andCallFake((size) => {
```

Comment:
As mentioned earlier, you don't need to create a spy here because `.createSpyObj()` does that for you.